### PR TITLE
Introduce util.SecondsSince instead of ad-hoc since()

### PR DESCRIPTION
**Pull request by pavelkalinnikov**
Forked from comment in #962.

### CODE REVIEW

File: util/timesource.go [hunk 1]
```diff
@@ -34,6 +37,10 @@ func (s SystemTimeSource) Now() time.Time {
 	return time.Now()
 }
 
+func (s SystemTimeSource) Since(t time.Time) time.Duration {
+	return s.Now().Sub(t)
```
Comment (pavelkalinnikov) [inline util/timesource.go]: Could probably use time.Since(t) instead.

**Comment by pavelkalinnikov** [comment]
@daviddrysdale Do you know other places where this method could be used?

**Comment by gdbelvin** [comment]
Thanks for making this change!  There is a linter in megacheck for this kind of thing. 
You might have to try a few of the linters but "simple", "golint", and "megacheck" are good candidates.

**Comment by pavelkalinnikov** [comment]
@gdbelvin Thanks for the tip, golint caught the same issues for me.
Also, will add a test shortly.

**Comment by daviddrysdale** [comment]
`monitoring/rpc_stats_interceptor.go` has a couple of similar usages (which also want `.Seconds()` btw); have a look and see if you feel it's a net gain in readability overall.

(Despite suggesting it, I'm on the fence -- the convenience might be outweighed by the unpleasantness of having a non-essential entrypoint in the `timeSource` interface).

**Comment by Martin2112** [comment]
Not sure how useful this really is but we can give it a go.

**Pull request merged**